test/session-pool.js

```diff
@@ -16,2022 +16,1575 @@
-          setTimeout(resolve, 500);
-        });
-      };
+      it('should release the session after calling the method', function(done) {
+        sessionPool.release = function(session) {
+          assert.strictEqual(session, SESSION);
+          done();
+        };
+
+        sessionPool.request_ = function(config, callback) {
+          callback();
+        };
 
-      return sessionPool.fill().then(function() {
-        assert(isAround(500, end()));
+        sessionPool.request(CONFIG, assert.ifError);
       });
-    });
-  });
 
-  describe('getLeaks', function() {
-    var formatTrace_;
+      it('should execute the callback with original arguments', function(done) {
+        const originalArgs = ['a', 'b', 'c'];
 
-    before(function() {
-      formatTrace_ = SessionPool.formatTrace_;
-    });
+        sessionPool.request_ = function(config, callback) {
+          callback.apply(null, originalArgs);
+        };
 
-    after(function() {
-      SessionPool.formatTrace_ = formatTrace_;
+        sessionPool.request(CONFIG, function() {
+          const args = [].slice.call(arguments);
+          assert.deepEqual(args, originalArgs);
+          done();
+        });
+      });
     });
 
-    it('should return a list of leaks', function() {
-      var fakeTraces = ['abc', 'def'];
-
-      sessionPool.traces_ = new Map(
-        fakeTraces.map(function(t, i) {
-          return [i, t];
-        })
-      );
+    describe('requestStream', function() {
+      let CONFIG;
+      let REQUEST_STREAM;
 
-      var formatCallCount = 0;
-      SessionPool.formatTrace_ = function(trace) {
-        assert.strictEqual(trace, fakeTraces[formatCallCount++]);
-        return trace
-          .split('')
-          .reverse()
-          .join('');
+      const SESSION = {
+        formattedName_: 'formatted-name',
       };
 
-      var traces = sessionPool.getLeaks();
+      beforeEach(function() {
+        REQUEST_STREAM = through();
 
-      assert.deepEqual(traces, ['cba', 'fed']);
-    });
-  });
+        CONFIG = {
+          reqOpts: {},
+        };
 
-  describe('getSession', function() {
-    it('should call through to acquireSession_', function() {
-      var fakeSession = {};
+        sessionPool.requestStream_ = function() {
+          return REQUEST_STREAM;
+        };
 
-      sessionPool.acquireSession_ = function(type) {
-        assert.strictEqual(type, 'readonly');
-        return Promise.resolve(fakeSession);
-      };
+        sessionPool.getReadSession = function() {
+          return Promise.resolve(SESSION);
+        };
 
-      return sessionPool.getSession().then(function(session) {
-        assert.strictEqual(session, fakeSession);
+        sessionPool.release = common.util.noop;
       });
-    });
-  });
 
-  describe('getWriteSession', function() {
-    it('should call through to acquireSession_', function() {
-      var fakeSession = {};
+      it('should get a session when stream opens', function(done) {
```

Comment:
I think you can remove this test, it's kind of unclear what is being tested here and it will be inherently tested elsewhere.

Reply:
This is not a new test case. It is copied from existing test cases for session pool. If you want, I can remove them.